File: src/ReactNativeSiriModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class ReactNativeSiriModule extends NativeModule {
  getAppGroup(): string | null;
  syncEntities(collection: string, items: Record<string, string>[]): Promise<void>;
  getSharedData(key: string): Promise<string | null>;
  setSharedData(key: string, value: string | null): Promise<void>;
  updateShortcuts(): Promise<void>;
  donateUserActivity(options: {
    activityType: string;
    title: string;
    userInfo: Record<string, string>;
    keywords: string[];
    persistentIdentifier: string | null;
    eligibleForSearch: boolean;
    eligibleForPrediction: boolean;
  }): Promise<void>;
  clearUserActivity(): Promise<void>;
}

export default requireNativeModule<ReactNativeSiriModule>('ReactNativeSiri');

File: android/src/main/java/expo/modules/reactnativesiri/ReactNativeSiriModule.kt
package expo.modules.reactnativesiri

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Siri App Intents are an iOS-only feature. This stub keeps the JS API safe
 * to call on Android: every function resolves as a no-op.
 */
class ReactNativeSiriModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ReactNativeSiri")

    Function("getAppGroup") {
      null
    }

    AsyncFunction("syncEntities") { _: String, _: List<Map<String, String>> ->
      null
    }

    AsyncFunction("getSharedData") { _: String ->
      null
    }

    AsyncFunction("setSharedData") { _: String, _: String? ->
      null
    }

    AsyncFunction("updateShortcuts") {
      null
    }

    AsyncFunction("donateUserActivity") { _: Map<String, Any?> ->
      null
    }

    AsyncFunction("clearUserActivity") {
      null
    }
  }
}
